### ISSUE TITLE
Reproducible random integer/real numbers are not obtained after setting seed

### ISSUE THREAD

**Issue by ivan-marroquin** [numpy/numpy#22381]
### Describe the issue:

Hi,

Thanks for such great package!

I have a windows 10 with Python 3.8.10, and I am using numpy 1.19.5.

I used these methods to fix the seed:
np.random.seed(1969)
rng= np.random.default_rng(1969)
seed= np.random.RandomState(1969)

to generate random integer/real numbers. Every time the I called the number generator twice, I get two different outputs. I ran the code below from a DOS terminal window. At each trial, I did the following: (1) open a new DOS terminal window, (2) run Python code, (3) close DOS terminal window.

Thanks for your help,
Ivan

### Reproduce the code example:

```python
import numpy as np
np.random.seed(1969)
np.random.rand(4)
array([0.59899771, 0.57768637, 0.92564433, 0.13716928])
np.random.rand(4)
array([0.09474638, 0.62259177, 0.92807496, 0.37224776])

import numpy as np
rng= np.random.default_rng(1969)
rng.random(4)
array([0.47441343, 0.82330681, 0.04106496, 0.32485782])
rng.random(4)
array([0.22002624, 0.9901044 , 0.0807824 , 0.00686858])

import numpy as np
seed= np.random.RandomState(1969)
seed.rand(4)
array([0.59899771, 0.57768637, 0.92564433, 0.13716928])
seed.rand(4)
array([0.09474638, 0.62259177, 0.92807496, 0.37224776])

import numpy as np
rng= np.random.default_rng(1969)
rng.integers(0,100,100)
array([ 5, 47,  3, 82, 58,  4, 77, 32, 60, 22, 33, 99, 32,  8, 57,  0,  5,
        9, 90, 24, 99, 36, 50, 16, 88, 32, 22, 24, 63, 70, 16, 67, 60, 83,
       45, 62, 66, 87, 92, 97, 77, 46, 51, 49, 37, 38, 77, 49,  7, 78, 41,
       43, 86, 54, 22,  4, 20, 32, 94, 50, 34, 11, 73, 69, 30, 81, 76,  0,
       66, 28, 95, 73, 43,  1, 71, 72, 49, 71, 21, 88, 51, 86,  3, 47, 36,
       83, 97, 72, 27, 18, 62, 99, 63, 48, 72, 73, 15, 79, 92, 82],
      dtype=int64)
rng.integers(0,100,100)
array([80, 10,  4, 34, 27, 37, 62, 69, 80, 80, 99,  7, 60, 30, 55, 81, 15,
       12, 31,  7,  8, 92, 58,  8, 54, 31,  5, 21, 41, 32, 80, 72, 25, 60,
       30, 19, 72,  8, 28,  8, 91, 25, 40, 19, 62, 94, 11, 20, 27, 31, 33,
       99, 14, 77, 75, 18, 12, 76, 99, 56, 21, 26, 87, 53, 95, 66, 41, 60,
       67, 61, 62, 86, 91, 97, 31, 57, 82, 16, 86, 47, 36, 50, 16, 74, 41,
       35, 51,  2, 38, 92, 24, 92, 86, 10, 22, 43, 58, 36, 47, 41],
      dtype=int64)
```

### Error message:

```shell
No error message was generated
```

### NumPy/Python version information:

print(np.__version__, sys.version)
1.19.5 3.8.10 (tags/v3.8.10:3d8993a, May  3 2021, 11:48:03) [MSC v.1928 64 bit (AMD64)]

**Comment by rkern** [comment]
This is expected behavior. `RandomState` and `Generator` are stateful. Calling the methods draws pseudorandom numbers from the stream, advancing the state of the underlying PRNG. Using a seeded `Generator` allows for reproducible _sequences of method calls_. Generally, you will get a seeded `Generator` _once_ at the start of your program and pass it around, calling methods on it wherever you need pseudorandom numbers.

```                                                                                                 
|1> rng = np.random.default_rng(0x6e11e0b05bd337497807b552da13ecbe)                                       
                                                                                                 
|2> rng.integers(0, 100, 10)                                                                              
array([19,  8, 90, 29,  3, 40, 99, 19, 92, 30])
                                                                                          
|3> rng.integers(0, 100, 10)                                                                              
array([98, 44,  1, 36, 47, 65, 50, 39, 41, 91])
                                                                                           
|4> rng = np.random.default_rng(0x6e11e0b05bd337497807b552da13ecbe)                                       
                                                                                           
|5> rng.integers(0, 100, 10)                                                                              
array([19,  8, 90, 29,  3, 40, 99, 19, 92, 30])
                                                                                              
|6> rng.integers(0, 100, 10)                                                                              
array([98, 44,  1, 36, 47, 65, 50, 39, 41, 91])
```

Most PRNG systems work like this. There are a few exceptions, like JAX, but they are usually [very clear about how they are different from most PRNG systems](https://jax.readthedocs.io/en/latest/jax-101/05-random-numbers.html).
